platform/mbed_wait_api_rtos.cpp

```diff
@@ -25,11 +25,18 @@
 #include "platform/mbed_power_mgmt.h"
 
 void wait(float s) {
-    wait_us(s * 1000000.0f);
+    wait_ms(s * 1000.0f);
 }
 
 void wait_ms(int ms) {
-    wait_us(ms * 1000);
+    // Shouldn't be calling a wait function from a critical section,
+    // but in case someone does...
+    if (core_util_are_interrupts_enabled()) {
```

Comment:
Wait_us class thread::wait (no busy wait), if the input is a multiple of 1000, which it is if the input is in milliseconds. So the behavior of your implementation is the same as in the current one besides that you are not surrounding it with sleep_manager_lock_deep_sleep - sleep_manager_unlock_deep_sleep.